SYCL/SubGroup/helper.hpp

```diff
@@ -169,5 +169,14 @@ bool core_sg_supported(const device &Device) {
   auto Vec = Device.get_info<info::device::extensions>();
   if (std::find(Vec.begin(), Vec.end(), "cl_khr_subgroups") != std::end(Vec))
     return true;
-  return Device.get_info<info::device::version>() >= "2.1";
+
+  if (Device.get_backend() == sycl::backend::opencl) {
```

Comment:
why is this not internal to opencl plugin?

Reply:
Because the OpenCL backend defines `info::device::version` as just passing through the whole version string from OpenCL, see:

* https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_backend_specific_information_descriptors

Reply:
I mean that plugin would check native version and return if "cl_khr_subgroups" extension is supported or not.
Similar to here: https://github.com/intel/llvm/blob/9008a5d28110f0fb847907ea4c8d2d5fe7af702b/sycl/plugins/opencl/pi_opencl.cpp#L609

Reply:
Oh I see what you mean, we could maybe do that, but I'm not sure it's quite correct, looking into it, I believe `cl_khr_subgroups` is now core in SYCL2020, so all devices should support it, and the only thing we could check in theory is if the sub-group size is `1`, so maybe we could remove/simplify this a lot, but I'm not sure all the plugins already implement this correctly so it would need some testing.

Could we leave it as-is in this PR that's just changing the version number, and I'll look into updating that in a follow-up?

Reply:
ok, sure